### PR TITLE
Don't set PYTHONPATH for :python3

### PR DESCRIPTION
**Pull request by tdsmith**
Reduces the impact of #43724 by not changing anything in the python2 case.

### CODE REVIEW

File: Library/Homebrew/requirements/python_requirement.rb [hunk 1]
```diff
@@ -24,7 +24,9 @@ class PythonRequirement < Requirement
       ENV.prepend_path "PATH", Formula["python"].opt_bin
     end
 
-    ENV["PYTHONPATH"] = "#{HOMEBREW_PREFIX}/lib/python#{short_version}/site-packages"
+    if python_binary == "python"
```
Comment (xu-cheng) [inline Library/Homebrew/requirements/python_requirement.rb]: I think this will always be true.

**Comment by mikemcquaid** [comment]
:+1: CC @dunn

File: Library/Homebrew/requirements/python_requirement.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Homebrew/requirements/python_requirement.rb]: The `env` isn't inherited, right? That's probably the cause of the confusion.

Comment (xu-cheng) [inline Library/Homebrew/requirements/python_requirement.rb]: Right, all of DSL wouldn't be able to be inherited by subclass.

Comment (mikemcquaid) [inline Library/Homebrew/requirements/python_requirement.rb]: Maybe it's worth refactoring this into a method and calling it from both pythons `env do` block. I believe that will sort all the problems as well as ensuring it's at the front of the `PATH`.

**Pull request closed without merging**